examples/03_connectivity/plot_rest_clustering.py

```diff
@@ -30,130 +51,148 @@
+# time with 1000 clusters. To see the power of joblib caching for second time.
+
+# We initialize class again with n_parcels=2000 this time.
 start = time.time()
-ward = FeatureAgglomeration(n_clusters=2000, connectivity=connectivity,
-                            linkage='ward', memory='nilearn_cache')
-ward.fit(fmri_masked)
+ward = Parcellations(method='ward', n_parcels=2000,
+                     standardize=False, smoothing_fwhm=2.,
+                     memory='nilearn_cache', memory_level=1,
+                     verbose=1)
+ward.fit(dataset.func)
 print("Ward agglomeration 2000 clusters: %.2fs" % (time.time() - start))
 
-##################################################################
-# Visualize results
-# ------------------
+###########################################################################
+# Visualize: Brain parcellations (Ward)
+# -------------------------------------
 #
-# First we display the labels of the clustering in the brain.
-#
-# To visualize results, we need to transform the clustering's labels back
-# to a neuroimaging volume. For this, we use the NiftiMasker's
-# inverse_transform method.
-from nilearn.plotting import plot_roi, plot_epi, show
+# First, we display the parcellations of the brain image stored in attribute
+# `labels_img_`
+ward_labels_img = ward.labels_img_
 
-# Unmask the labels
+# Now, ward_labels_img are Nifti1Image object, it can be saved to file
+# with the following code:
+ward_labels_img.to_filename('ward_parcellation.nii')
 
-# Avoid 0 label
-labels = ward.labels_ + 1
-labels_img = nifti_masker.inverse_transform(labels)
+from nilearn import plotting
+from nilearn.image import mean_img, index_img
 
-from nilearn.image import mean_img
-mean_func_img = mean_img(func_filename)
+# we take mean over time on the functional image to use mean image as
+# background to parcellated image assigned to ward_labels_img
+mean_func_img = mean_img(dataset.func[0])
 
+first_plot = plotting.plot_roi(ward_labels_img, mean_func_img,
+                               title="Ward parcellation",
+                               display_mode='xz')
 
-first_plot = plot_roi(labels_img, mean_func_img, title="Ward parcellation",
-                      display_mode='xz')
+###########################################################################
+# Compressed representation of Ward clustering
+# --------------------------------------------
+#
+# Second, we illustrate the effect that the clustering has on the signal.
+# We show the original data, and the approximation provided by the
+# clustering by averaging the signal on each parcel.
+
+# grab number of voxels from attribute mask image. We use sum operator along
+# with math_img from nilearn
+from nilearn.image import math_img
+
+original_voxels = math_img("np.sum(img)", img=ward.mask_img_).get_data()
+
+# common vmin and vmax
+vmin = math_img("np.min(img)", img=mean_func_img).get_data()
+vmax = math_img("np.max(img)", img=mean_func_img).get_data()
```

Comment:
I think that it would be better to not use math_img and simply run np.sum, np.min, and np.max on the result of .get_data() above.